sql/sqlstats/sslocal: TestSQLStatsDiscardStatsOnFingerprintLimit failed

sql/sqlstats/sslocal.TestSQLStatsDiscardStatsOnFingerprintLimit [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20957731?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20957731?buildTab=artifacts#/) on master @ [df498b8b1a9ea5e9520f7aec89bed6552c013a0e](https://github.com/cockroachdb/cockroach/commits/df498b8b1a9ea5e9520f7aec89bed6552c013a0e):


```
=== RUN   TestSQLStatsDiscardStatsOnFingerprintLimit
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/5e8d9296e8c352d3ee30298955d1f8e0/logTestSQLStatsDiscardStatsOnFingerprintLimit890725023
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:181: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    test_server_shim.go:666: DRPC is enabled (via override by TestingGlobalDRPCOption)
    sql_stats_test.go:1799: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5e8d9296e8c352d3ee30298955d1f8e0/logTestSQLStatsDiscardStatsOnFingerprintLimit890725023
--- FAIL: TestSQLStatsDiscardStatsOnFingerprintLimit (9.53s)
=== RUN   TestSQLStatsDiscardStatsOnFingerprintLimit/statement_and_txn_limit_reached_across_applications
    sql_stats_test.go:1787: condition failed to evaluate within 5s: from testutils.go:512: expected 1 transaction(s), got 0 filters=
--- FAIL: TestSQLStatsDiscardStatsOnFingerprintLimit/statement_and_txn_limit_reached_across_applications (5.74s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLStatsDiscardStatsOnFingerprintLimit.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-58571

--- a/pkg/sql/sqlstats/sslocal/sql_stats_test.go
+++ b/pkg/sql/sqlstats/sslocal/sql_stats_test.go
@@ -1781,10 +1781,10 @@ func TestSQLStatsDiscardStatsOnFingerprintLimit(t *testing.T) {
 					sqlstatstestutil.StatementFilter{AllowInternal: true})
 			}
 			if tc.txnLimit == 0 {
-				sqlstatstestutil.WaitForTransactionEntriesAtLeast(t, utilConn, tc.minTxns,
-					sqlstatstestutil.TransactionFilter{AllowInternal: true})
+				sqlstatstestutil.WaitForTransactionEntriesAtLeast(t, utilConn, tc.minTxns)
 			} else {
-				sqlstatstestutil.WaitForTransactionEntriesEqual(t, utilConn, tc.txnLimit-1)
+				sqlstatstestutil.WaitForTransactionEntriesEqual(t, utilConn, tc.txnLimit-1,
+					sqlstatstestutil.TransactionFilter{AllowInternal: true})
 			}
 			testutils.SucceedsSoon(t, func() error {
 				if discardedMetric.Count() < int64(tc.totalSkipped) {

